app/controllers/application_controller.rb

```diff
@@ -1,10 +1,153 @@
 require './config/environment'
-
+require "pry"
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "time again"
+  end
+
+  get "/" do
+    erb :index
+  end
+
+  get "/signup" do
+    if logged_in?
+      redirect "/tweets"
+    else
+      erb :"users/create_user"
+    end
+  end
+
+  post "/signup" do
+    if !params[:username].empty? && !params[:email].empty? && !params[:password].empty?
+      @user = User.create(params)
+      session[:id] = @user.id
+
+      redirect "/tweets"
+    else
+      redirect "/signup"
+    end
+  end
+
+  get "/login" do
+    if logged_in?
+      redirect "/tweets"
+    else
+      erb :"users/login"
+    end
+  end
+
+  post "/login" do
+    @user = User.find_by(username: params[:username])
+    if @user && @user.authenticate(params[:password])
+      session[:id] = @user.id
+      redirect "/tweets"
+    else
+      redirect "/login"
+    end
+  end
+
+  get "/tweets/logout" do
+    session.clear
+    redirect "/"
+  end
+
+  get "/tweets/new" do
+    if logged_in?
+      erb :"tweets/create_tweet"
+    else
+      redirect "/login"
+    end
+  end
+
+  post "/tweets" do
+    if !params[:content].empty?
+      @tweet = Tweet.create(content: params[:content], user_id: current_user.id)
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.